### ISSUE TITLE
Visualize transaction data in confirmation dialog 

### ISSUE THREAD

**Issue by mobinauten** [WICG/digital-credentials#195]
The EU LSP projects NOBID, EWC implemented SCA and payment initiation via OIDVP. 
SCA is btw. mandatoryfo rmany parties bacled by eiDAS2. 
This new API/protocol should also visualize transaction data as defined in https://openid.net/specs/openid-4-verifiable-presentations-1_0.html#name-transaction-data

The payment and SCA flows are described here - https://github.com/nobid-consortium/payment-reference-documentation/blob/main/payment-reference-doc.md

**Comment by leecam** [comment]
I think the responsibility for visualizing this transaction data belongs to the wallet, not the browser. So I don't think the DC API spec needs to concern itself directly with transaction data that maybe included in any given presentation protocol.

That said platforms may offer wallets native API to optimize this UX flow. Indeed Android provides Wallets an API to visualize payment transaction data directly onto to selector screen. This is purely a UX optimization, the wallet app is still the one responsible for parsing the transaction data and displaying it. To give you an idea the screenshot below is what the developer preview UX looks like on Android today if the wallet detects transaction data in the request.

Ultimately though this is the responsibly of the wallet. The platforms are free to optimize this how they wish (if at all), not something we'd mandate at the DC API level. 

![image](https://github.com/user-attachments/assets/6ac00267-6b54-4789-8a6d-244541837463)

**Comment by mobinauten** [comment]
Yes, the Wallet should cover most of the payment specifis within their visualization, I agree,  as prototyped in the 1st screen attached 
![IMG_3002](https://github.com/user-attachments/assets/a344c1f6-324d-4075-8828-220bfa92e307) but the "Wallet Selection Screen" _could_ already show not only what kind of credential but for what transaction data the wallet should be used. I agree one doesn't need to be, it's a CX/comfortability feature (see second attached image, the Wallet Selector). 
![Image](https://github.com/user-attachments/assets/1e289b5d-6e56-46ad-9170-6f141d2a9eb3). 
How one use this payment screen attached in the browser? Probably then too many screens, steps. Just thinking?

**Comment by leecam** [comment]
Yep agree.

The screenshot I posted above is Android doing what you are suggesting. Its optimizing the selection UI screen (your 2nd screenshot) to show the transaction data directly. This potentially saves the wallet from having to do it in a follow up screen, saving a click and making it feel more like payment confirmation. This is just a platform specific UX optimization though, not something we can spec at the browser API level at this point I don't think.

**Comment by mobinauten** [comment]
As suggested I am currently adding the traditional Payment Screen to the Android Wallet in see how it feels.  That is not an issue I assume. I believe we'll find a suitable way. Sill not sure if we not should visualize the transaction data on the Wallet selection screen. This would not only help payment but maybe also QES use cases?

**Comment by timcappalli** [comment]
@mobinauten the credential selector (what you referred to as the wallet selection screen) is not something that is defined by the Digital Credentials API specification.

**Comment by mobinauten** [comment]
Ah, Understood. Thx. Who then? The Credential Manager API?

**Comment by timcappalli** [comment]
The credential selector is a platform capability.

**Comment by timcappalli** [comment]
Not related to the web platform Digital Credential API
